ios-base/Classes/ErrorHandler.m

```diff
@@ -0,0 +1,137 @@
+//
+//  ErrorHandler.m
+//
+
+#import "ErrorHandler.h"
+
+
+#pragma mark -
+@implementation ErrorHandler
+
++ (void)errorDescription:(NSDictionary *)errorDict
+                     key:(NSString **)resultKey
+                  reason:(NSString **)resultReason
+{
+    if ([errorDict isKindOfClass:[NSDictionary class]])
+    {
+        id errors = [errorDict objectForKey:@"errors"];
+        if (errors)
+        {
+            errorDict = errors;
+        }
+    }
+    
+    for (NSString *key in errorDict)
+    {
+        if ([key isEqualToString:@"status"])
+        {
+            continue;
+        }
+        
+        NSString *value     = [errorDict objectForKey:key];
+        if ([value isKindOfClass:[NSArray class]])
+        {
+            value           = [(NSArray *)value componentsJoinedByString:@", "];
+        }
+        
+        if ([value isKindOfClass:[NSString class]])
+        {
+            if (resultKey)
+            {
+                *resultKey            = key;
+            }
+            
+            if (resultReason)
+            {
+                *resultReason         = value;
```

Comment:
@nikitafomin you about lines or space between "resultKey" and "key"?
I don't like so much space between var and value if there is not a lot of values together
But lines is OK.